frappe/public/js/frappe/form/quick_entry.js

```diff
@@ -35,7 +35,11 @@ frappe.ui.form.QuickEntryForm = Class.extend({
 				if (this.is_quick_entry() || this.force) {
 					this.render_dialog();
 					resolve(this);
-				} else {
+				} else { // No quick entry, use full Form
+					// but still give callback a shot at the doc
+					if (this.init_callback) {
+						this.init_callback(this.doc);
+					}
```

Comment:
- Use `if-else if-else` for this, rather than checking callback in else

Reply:
Sorry I am not following; if the condition on line 35 is true, we render the dialog (which ultimately will call the callback); but if not, then we have to set the route. In the else clause from that condition on line 35 is our only opportunity to call the callback along that code path, but of course we can't call it if it's not defined. So we need to check that. So I am not sure how you'd like me to restructure the code. I am happy to change it however you like, I am just not clear on exactly what you propose. I look forward to further detail from you, thanks.